### PR TITLE
Fixes emptied shot glasses having the drinking glass filling icon.

### LINKED ISSUE #62904: Emptied shot glasses use drinking glasses' overlay
## Reproduction:
right so you know how when you pour a glass of anything in a fresh glass to then empty it however and then the glass appears a little dirtied and smudged? it's making an overlay with no base colors since there are no reagents in the glass to take the colors from, so it defaults to a grey-ish white

shot glasses do that as well and they've been intended to have their own separate overlay for this but the code is fucked somewhere and it doesn't even get to that point and just slaps a drinking glass overlay instead, resulting in these
image

relevant code:

https://github.com/tgstation/tgstation/blob/b722c89d5f75afdd9469a049018b1dec14e42d9a/code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm#L37-L54

https://github.com/tgstation/tgstation/blob/b722c89d5f75afdd9469a049018b1dec14e42d9a/code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm#L90-L106

of note is that some drinks have their shot_glass_icon_state set to shotglassclear, resulting in their unique overlay being overridden by the colored one, i.e. vodka shots are supposed to be clear but are instead blue

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Made both drinking and shot glasses use the generic `reagent_containers/update_overlay()` method. Standardized drinking and shot glasses `update_icon_state()`.

## Why It's Good For The Game
This will fix #62904 and remove some bad code for drinking glasses overlays.

## Changelog

:cl:
fix: Fixed emptied shot glasses having the drinking glass filling icon.
/:cl:

### CODE REVIEW

**Comment by carshalash** [comment]
Soooo, no more garbage grey overlay for empty glasses?

**Comment by Ghommie** [comment]
> 
> 
> Soooo, no more garbage grey overlay for empty glasses?

Correct.

**Comment by carshalash** [comment]
> > Soooo, no more garbage grey overlay for empty glasses?
> 
> Correct.

I love you

**Comment by GoldenAlpharex** [comment]
Amazing, wonderful even.

**Review by Twaticus — APPROVED** [review]

File: code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm [hunk 1]
```diff
@@ -36,22 +37,21 @@
...
-	var/datum/reagent/largest_reagent = reagents.get_master_reagent()
-	if(largest_reagent?.glass_icon_state)
-		icon_state = largest_reagent.glass_icon_state
+	var/glass_icon = get_glass_icon(reagents.get_master_reagent())
+	if(glass_icon)
+		icon_state = glass_icon
+		fill_icon_thresholds = null
+	else
+		//Make sure the fill_icon_thresholds and the icon_state are reset. We'll use reagent overlays.
+		fill_icon_thresholds = fill_icon_thresholds || list(1)
```
Comment (Watermelon914) [inline code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm:49]: maybe set to initial instead?

**Comment by carshalash** [comment]
VERY broken. certain things will be displayed in shot glasses instead of regular glasses, cocktail unique sprites are gonna and only the hex value is displaying in its place.

**Pull request merged**